persist/fs/clone/cloner_test.go

```diff
@@ -87,14 +107,32 @@ func TestCloner(t *testing.T) {
 	}
 	require.NoError(t, r2.Open(r2OpenOpts))
 	for {
-		t1, b1, c1, e1 := r1.Read()
-		t2, b2, c2, e2 := r2.Read()
+		t1, a1, b1, c1, e1 := r1.Read()
+		t2, a2, b2, c2, e2 := r2.Read()
 		if e1 == e2 && e1 == io.EOF {
 			break
 		}
+
+		require.NoError(t, e1)
+		require.NoError(t, e2)
+
 		b1.IncRef()
 		b2.IncRef()
 		require.Equal(t, t1.String(), t2.String())
+		require.Equal(t, a1.Remaining(), a2.Remaining())
+		numTags, numTagsMatched := a1.Remaining(), 0
```

Comment:
there's a matcher for tag iters, so can make this a little shorter if you'd like:

```
 ident.NewTagIterMatcher(ident.NewTagSliceIterator(a1)).Matches(ident.NewTagSliceIterator(a2)))
```

Reply:
Ah nice, will update - gracias.